### PR TITLE
snagboot: init at 1.0

### PR DESCRIPTION
**Pull request by otavio**
###### Description of changes

Requires:

- https://github.com/NixOS/nixpkgs/pull/234534
- https://github.com/NixOS/nixpkgs/pull/234527

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/snagboot/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    fi
+
+    mkdir -p "$out/lib/udev/rules.d"
+    cp "$rules" "$out/lib/udev/rules.d/80-snagboot.rules"
+  '';
+
+  # There are no tests
+  doCheck = false;
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/applications/misc/snagboot/default.nix:69]:
```suggestion
  meta = {
```

File: pkgs/applications/misc/snagboot/default.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    homepage = "https://github.com/bootlin/snagboot";
+    license = licenses.gpl2;
+    description = "Generic recovery and reflashing tool for embedded platforms";
+    maintainers = with maintainers; [ otavio ];
```
Comment (AndersonTorres) [inline pkgs/applications/misc/snagboot/default.nix:73]:
```suggestion
    homepage = "https://github.com/bootlin/snagboot";
    description = "Generic recovery and reflashing tool for embedded platforms";
    license = lib.licenses.gpl2;
    maintainers = with lib.maintainers; [ otavio ];
```

File: pkgs/applications/misc/snagboot/default.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+, testers
+, gitUpdater
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "snagboot";
+  version = "1.0";
+  format = "pyproject";
+
+  src = python3.pkgs.fetchPypi {
```
Comment (AndersonTorres) [inline pkgs/applications/misc/snagboot/default.nix:15]: I believe it is being deprecated in favor of the toplevel one. But keep it for now.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/snagboot/default.nix (hunk 3 above)
Comment (otavio) [reply pkgs/applications/misc/snagboot/default.nix:15]: I used this because we mix python3 and non-python pkgs below.

Comment (otavio) [reply pkgs/applications/misc/snagboot/default.nix]: Fixed.

File: pkgs/applications/misc/snagboot/default.nix [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "snagboot";
+  version = "1.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-wtIcrd3s/ZfdYqi2a2+IvVYnJie5txJy6d2m+GjuhxU=";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/snagboot/default.nix:18]:
```suggestion
    hash = "sha256-wtIcrd3s/ZfdYqi2a2+IvVYnJie5txJy6d2m+GjuhxU=";
```

**Review by AndersonTorres — APPROVED** [review]
besides lgtm

**Pull request merged**